orderbook/src/price_estimate.rs

```diff
@@ -68,33 +167,95 @@ mod tests {
     async fn estimate_price_on_direct_pair() {
         let token_a = H160::from_low_u64_be(1);
         let token_b = H160::from_low_u64_be(2);
-        let pool = Pool::uniswap(TokenPair::new(token_a, token_b).unwrap(), (100, 10));
+        let pool = Pool::uniswap(
+            TokenPair::new(token_a, token_b).unwrap(),
+            (10u128.pow(30), 10u128.pow(29)),
+        );
 
         let pool_fetcher = Box::new(FakePoolFetcher(vec![pool]));
-        let estimator = UniswapPriceEstimator { pool_fetcher };
+        let estimator = UniswapPriceEstimator::new(pool_fetcher, hashset!());
 
         assert_approx_eq!(
-            estimator.estimate_price(token_a, token_a).await.unwrap(),
+            estimator
+                .estimate_price(token_a, token_a, U256::exp10(18), OrderKind::Buy)
+                .await
+                .unwrap(),
+            1.0
+        );
+        assert_approx_eq!(
+            estimator
+                .estimate_price(token_a, token_a, U256::exp10(18), OrderKind::Sell)
+                .await
+                .unwrap(),
             1.0
         );
```

Comment:
I like these tests. price of X relative to X is 1 no matter if you're buying or selling.